File: frontend/src/lib/api/theme.ts
import { apiClient } from "@/lib/api/client";

export interface ThemeTokens {
  name: string;
  description: string;
  palette_type: "categorical" | "sequential" | "diverging";
  background: string;
  foreground: string;
  grid: string;
  border: string;
  categorical_colors: string[];
  sequential_range: [string, string];
  diverging_range: [string, string, string];
  positive_color: string;
  negative_color: string;
  headline_font: string;
  body_font: string;
}

export interface ThemeRecommendationsResponse {
  top: ThemeTokens[];
  rest: ThemeTokens[];
}

export function listThemes() {
  return apiClient.get<ThemeTokens[]>("/api/themes");
}

export function getThemeRecommendations() {
  return apiClient.get<ThemeRecommendationsResponse>("/api/themes/recommendations");
}

File: backend/app/schemas/theme.py
from pydantic import BaseModel


class ThemeResponse(BaseModel):
    name: str
    description: str
    palette_type: str
    background: str
    foreground: str
    grid: str
    border: str
    categorical_colors: list[str]
    sequential_range: tuple[str, str]
    diverging_range: tuple[str, str, str]
    positive_color: str
    negative_color: str
    headline_font: str
    body_font: str


class ThemeRecommendationsResponse(BaseModel):
    top: list[ThemeResponse]
    rest: list[ThemeResponse]
